core/arch/arch_exports.h

```diff
@@ -157,17 +157,30 @@ typedef struct _spill_state_t {
     reg_t reg_stolen; /* slot for the stolen register */
 #elif defined(RISCV64)
     reg_t a0, a1, a2, a3;
+    /* Slot for the "stolen" register.
+     * Note that on RISC-V, tp is the thread pointer, and it is also a general-purpose
+     * register, so unlike AARCHXX, we don't actually need to steal any register to store
+     * DR's tls, we can just use tp directly.
+     * But this also introduces another issue, we will lose the original value of
+     * tp. So in insert_load_dr_tls_base() we put the original value of tp into the x0
+     * slot of the dcontext, and in append_save_gpr() we restore the original value back
+     * to tp.
+     * We'll also need to mangle any instructions that use tp.
```

Comment:
If we use tp as the stolen register, it will be the only one we have to save & restore, I suppose it simplifies things, right?

Reply:
While in the code cache I don't think using tp is any different from another register (unless tp is used less frequently than any other register by app code).  The place where it may save work is transitioning back to DR system code if the same value is used for DR's C code.  But code cache operation is what should be optimized for so if some other register is used a lot less that might outweigh the cxt sw savings: cxt switches are already really expensive.  All that said: it seems fine to go with tp for now.

Reply:
I guess on a clean call too: that may be more frequent.